tests/gireg.lua

```diff
@@ -154,18 +174,28 @@ function gireg.type_int64()
 -- is always 'double', and conversion between double and int64 big
 -- constants is always lossy.  Not sure if it can be solved somehow.
 
---   checkv(R.test_int64(0x7fffffffffffffff), 0x7fffffffffffffff, 'number')
---   checkv(R.test_int64(-0x8000000000000000), -0x8000000000000000, 'number')
---   check(not pcall(R.test_int64, 0x8000000000000000))
---   check(not pcall(R.test_int64, -0x8000000000000001))
+-- UPDATE: It IS working in Lua 5.3 because of its integer support,
+-- so tests are enabled there.
+
+   if nativeIntegers then
+      checkv(R.test_int64(0x7fffffffffffffff), 0x7fffffffffffffff, 'number')
+      checkv(R.test_int64(-0x8000000000000000), -0x8000000000000000, 'number')
+   end
+
+   -- Following two lines succeed because integers wrap around in Lua 5.3 like they do in C.
```

Comment:
Integers *do not* wrap around in C. Please remove that remark.
(I wonder how Lua manages to get wrap-around integers... From a quick look at the implementation: It doesn't.)

Reply:
What am I missing?

```Lua
Lua 5.3.3  Copyright (C) 1994-2016 Lua.org, PUC-Rio
> 0x7fffffffffffffff
9223372036854775807
> 0x7fffffffffffffff+1
-9223372036854775808
```

They do wrap just fine…